From c2014b756f5646e3caa9f6c96031faf370d0dfd1 Mon Sep 17 00:00:00 2001
From: Huang Shijie <b32955@freescale.com>
Date: Mon, 1 Nov 2010 10:17:15 +0800
Subject: [PATCH] ENGR00133178-2 DMA : remove mutex for each channel

The spin_lock is enough for each DMA channel, so
remove the mutex lock.

Signed-off-by: Huang Shijie <b32955@freescale.com>
---
 arch/arm/mach-mx5/dmaengine.c |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-mx5/dmaengine.c b/arch/arm/mach-mx5/dmaengine.c
index b8aad6c..ff3bb35 100644
--- a/arch/arm/mach-mx5/dmaengine.c
+++ b/arch/arm/mach-mx5/dmaengine.c
@@ -128,13 +128,11 @@ int mxs_dma_enable(int channel)
 		return -EINVAL;
 
 	pdma = pchan->dma;
-	mutex_lock(&mxs_dma_mutex);
 	spin_lock_irqsave(&pchan->lock, flags);
 	if (pchan->pending_num && pdma->enable)
 		ret = pdma->enable(pchan, channel - pdma->chan_base);
 	pchan->flags |= MXS_DMA_FLAGS_BUSY;
 	spin_unlock_irqrestore(&pchan->lock, flags);
-	mutex_unlock(&mxs_dma_mutex);
 	return ret;
 }
 EXPORT_SYMBOL(mxs_dma_enable);
@@ -152,7 +150,6 @@ void mxs_dma_disable(int channel)
 	if (!(pchan->flags & MXS_DMA_FLAGS_BUSY))
 		return;
 	pdma = pchan->dma;
-	mutex_lock(&mxs_dma_mutex);
 	spin_lock_irqsave(&pchan->lock, flags);
 	if (pdma->disable)
 		pdma->disable(pchan, channel - pdma->chan_base);
@@ -161,7 +158,6 @@ void mxs_dma_disable(int channel)
 	pchan->pending_num = 0;
 	list_splice_init(&pchan->active, &pchan->done);
 	spin_unlock_irqrestore(&pchan->lock, flags);
-	mutex_unlock(&mxs_dma_mutex);
 }
 EXPORT_SYMBOL(mxs_dma_disable);
 
-- 
1.5.4.4

